File: programmers/2-42586-기능개발/junghwan.js
function solution(progresses, speeds) {
    var answer = [];
    var p = progresses;
    var s = speeds;
    var day = [];

    // 배포까지 남은 일수 계산
    for(let i=0; i<p.length; i++) {
        // (100-93) / 1
        let deployDay = 0;
        if((100-p[i]) % s[i] !== 0) {
            deployDay = parseInt((100-p[i])/s[i]) + 1;
        }
        else {
            deployDay = parseInt((100-p[i])/s[i]);
        }
        day.push(deployDay);
    } //[7,3,9]

    console.log(day);

    let cnt = 0;
    let idx = 0;
    let i = 0;

    while(idx < day.length) {
        cnt = 1; // 기본값은 1

        // 자기 자신 다음 값부터 차례대로 비교
        for(i=idx+1; i<day.length; i++) {
            if(day[idx] < day[i]) { // 다음 값이 크면
                answer.push(cnt); 
                break;
                // answer에 cnt 값 삽입 후 반복문 종료
            }
            
            // 작으면 배포일수 증가
            cnt++;
        }

        // 인덱스 변경
        idx = i;

        console.log(`cnt: ${cnt}`);
        console.log(`idx: ${idx}`);
        console.log('');
    }

    answer.push(cnt);

    //[7,9,3]
    return answer;
}

File: programmers/2-42586-기능개발/junghwan.py
# 기능개발

'''
19. 06. 20

풀이: day list 안에 배포까지 남은 일자 계산 후 append
반복문을 통해 대소 비교 후 cnt 값 증가 및 answer 리스트에 append

기존 js로 풀었던 방식으로 해결해보려 했으나, 비효율적이고 문법적으로도 이슈가 있었음
(인덱스 값을 다른 변수에 저장할 때 발생)

'''

import math

def solution(progresses, speeds):
    answer = []
    day = []

    for i in range(len(progresses)):
        day.append(math.ceil((100 - progresses[i]) / speeds[i]))

    idx = 0
    cnt = 1
    i = 1
    while i < len(day):

        if day[idx] < day[i]:
            idx = i
            answer.append(cnt)
            cnt = 0
        
        else:
            i = i + 1
            cnt = cnt + 1

    answer.append(cnt)
    print(answer)

    return answer